crates/iota-network/src/discovery/mod.rs

```diff
@@ -619,20 +843,24 @@ fn update_known_peers(
             continue;
         }
 
-        // Skip entries that have too many addresses as a means to cap the size of a
-        // node's info
-        if peer_info.addresses.len() > MAX_ADDRESSES_PER_PEER {
+        // Skip entries that have no or too many addresses as a means to cap the size of
+        // a node's info. This also means we don't update entries when peers remove all
+        // their addresses. Those entries will eventually be removed after
+        // ONE_DAY_MILLISECONDS.
+        if peer_info.addresses.is_empty() || peer_info.addresses.len() > MAX_ADDRESSES_PER_PEER {
             continue;
         }
 
-        // verify that all addresses provided are valid anemo addresses
+        // Verify that all addresses provided are valid anemo addresses
         if !peer_info
             .addresses
             .iter()
             .all(|addr| addr.len() < MAX_ADDRESS_LENGTH && addr.to_anemo_address().is_ok())
         {
             continue;
         }
+
+        // Verify the signature
```

Comment:
I see no reason for `debug_fatal!` two lines below. A malicious node can just span with invalid signatures. A node built in debug mode will crash, a release mode node will report system_invariant_violation although there's no violations here, we just normally verify the public key, and if it's invalid we just skip peer_info, no violation.

Reply:
Yeah I was thinking about the same line. I thought it's impossible to reach this `debug_fatal!` (this comes from upstream), but I guess someone can send signatures that can't be passed. I will make a `debug!` out of it.